runtimes/peregrine/src/xcm_config.rs

```diff
@@ -98,8 +108,16 @@ pub type XcmBarrier = DenyThenTry<
 /// parameters.
 pub struct SafeCallFilter;
 impl Contains<RuntimeCall> for SafeCallFilter {
-	fn contains(_call: &RuntimeCall) -> bool {
-		false
+	fn contains(call: &RuntimeCall) -> bool {
+		matches!(
+			call,
+			RuntimeCall::Did { .. }
+				| RuntimeCall::Ctype { .. }
+				| RuntimeCall::DidLookup { .. }
+				| RuntimeCall::Web3Names { .. }
+				| RuntimeCall::PublicCredentials { .. }
+				| RuntimeCall::Attestation { .. }
+		)
 	}
 }
```

Comment:
I whitelisted all relevant DID pallets. According to the comment, I guess I should only allow the `create_from_account` extrinsic in the did pallet.

Reply:
I think it's enough to ban anything recursive like `proxy` calls and `utility` batches.

Reply:
Alright okay. I think it is also fine. Do you want to allow more? Otherwise, I would resolve this conversation.

Reply:
It is a measure to limit PoV sizes, so really anything that does not have unbounded execution. As far as I know everything but nested calls are bounded, so it should be fine and future-proof.

The comment from the source code for v1.0.0 is:

```
/// Calls that are allowed through this filter must:
/// 1. Have a fixed weight;
/// 2. Cannot lead to another call being made;
/// 3. Have a defined proof size weight, e.g. no unbounded vecs in call parameters.
```

I am actually not sure why point 3 is there. AFAIK PoV is not affected by input length, but only by accessed storage size (see [here](https://substrate.stackexchange.com/questions/9264/what-is-proof-size)). Anyway, it makes sense to check if we take `Vec`s anywhere (we shouldn't), and exclude those calls as well.

Also, I read somewhere, next versions of Polkadot remove the need to have this filter configured, not sure which version that was tho.

Reply:
Yeah, but if we don't allow `dispatch_as`  the did pallets are not really useful. Other parachains can not use their DID to do any operation. 
 
 I think we have to implement our own origin converter which creates the double origin for specific runtime calls.

Reply:
Or rather I would only check that it calls one of `ctype`, `attestation`, `delegation`, `web3name`, and `did` management operations, to start with.

Reply:
the `SafecallFilter` is updated in[2387069](https://github.com/KILTprotocol/kilt-node/pull/603/commits/2387069acf5cfc76035ba8535f0e69e896dc04fc)